learn/security/tenant_tokens.md

```diff
@@ -0,0 +1,299 @@
+const yourSearchRules = {
+  "patient_medical_records": {
+    "filter": "user_id = 1"
+  }
+}
+```
+
+`searchRules` must be an object containing at least one search rule. A search rule must specify either an index name or a `*` wildcard. Search rules must also declare at least one filter. In this example, the filter makes it so a user only has access to their own records.
+
+With both these steps completed, you are ready to generate the token payload:
+
+```js
+const payload = {
+  "apiKeyPrefix": yourApiKeyPrefix,
+  "exp": 1641835850,
+  "searchRules": yourSearchRules
+}
+```
+
+You can also optionally set an expiry date for your token. Tokens are invalid once past their expiry date.
+
+Like the header, it is necessary to encode the payload:
+
+```js
+const base64Payload = base64Encode(payload);
+```
+
+You can read more about each of element of the tenant token payload in [this guide's final section](#tenant-token-payload-reference).
+
+### Step 3: Signing a token
+
+Tokens following the JWT standard must be signed. The signature is created by concatenating the encoded header and payload using a `.` as the separator, then encrypting the resulting string. **The encryption key must be the same you used when generating the payload's API key prefix.**
+
+```js
+const signature = HS256(base64Header + '.' + base64Payload, meiliSearchApiKey);
+```
+
+This example uses an HS256 encryption algorithm. **The encryption algorithm you use must be the same as the one indicated in the token's header.**
+
+The remaining step is putting it all together. Concatenate the header, payload, and signature using a `.` to separate each element of the token:
+
+```js
+const yourTenantToken = base64Header + '.' + base64Payload + '.' + signature;
+```
+
+### Using a tenant token without an SDK
+
+Your token is ready. Tenant tokens can seamlessly replace API keys to authorize requests made to the search endpoint:
+
+<CodeSamples id="tenant_token_guide_search_no_sdk_1" />
+
+::: note
+The `curl` example presented here is only for illustration purposes. In production environments, you would likely send the token to the front-end of your application and query indexes from there.
+:::
+
+## Tenant token payload reference
+
+Meilisearch's tenant tokens are JWTs. Their payload is made of three elements: [search rules](#search-rules), an [API key](#api-keys), and an optional [expiration date](#expiry-date).
+
+### Search rules
+
+Search rules are a set of instructions defining search parameters that will be enforced in every query made with a specific tenant token.
+
+`searchRules` must contain a JSON object specifying rules that will be enforced on any queries using this token. Each rule is itself a JSON object and must follow this pattern:
+
+```json
+{
+ "[index_name]": {
+   "[search_parameter]": "[value]"
+ }
+}
+```
+
+The object key must be an index name. You may also use `*` instead of a specific index name—in this case, search rules will be applied to all indexes.
```

Comment:
Maybe mention that * is a wildcard here because you mention overwriting wildcard rules in line 213? It maybe very obvious to some people, so Idk if its necessary